### PR TITLE
Support relative paths for PBR materials

### LINKED ISSUE #286: PBR file paths cannot be relative
Material maps within an SDF file require `model://`. For example:

```
          <pbr>
            <metal>
              <albedo_map>model://UnderwaterWorld/materials/textures/Rock_Albedo.png</albedo_map>
              <normal_map>model://UnderwaterWorld/materials/textures/Rock_Normal.png</normal_map>
              <roughness_map>model://UnderwaterWorld/materials/textures/Rock_Roughness.png</roughness_map>
            </metal>
          </pbr>
```

It would be nice if these paths could be relative like `<mesh><uri>`.

### PR DESCRIPTION
**Pull request by nkoenig**
Resolves #286.

Requires: https://github.com/osrf/sdformat/pull/349.

To test, you can modify PBR map URIs similar to the following.

Old
```
<pbr>
  <metal>
    <albedo_map>https://fuel.ignitionrobotics.org/1.0/OpenRobotics/models/Cave Straight Type A/3/files/materials/textures/Gravel_Albedo.png</albedo_map>
    <normal_map>https://fuel.ignitionrobotics.org/1.0/OpenRobotics/models/Cave Straight Type A/3/files/materials/textures/Gravel_Normal.png</normal_map>
    <roughness_map>https://fuel.ignitionrobotics.org/1.0/OpenRobotics/models/Cave Straight Type A/3/files/materials/textures/Gravel_Roughness.png</roughness_map>
  </metal>
</pbr>
```

New
```
<pbr>
  <metal>
    <albedo_map>materials/textures/Gravel_Albedo.png</albedo_map>
    <normal_map>materials/textures/Gravel_Normal.png</normal_map>
    <roughness_map>materials/textures/Gravel_Roughness.png</roughness_map>
  </metal>
</pbr>
```

Signed-off-by: Nate Koenig <nate@openrobotics.org>

### CODE REVIEW

File: src/Conversions.cc [hunk 1]
```diff
@@ -259,16 +259,25 @@ msgs::Material ignition::gazebo::convert(const sdf::Material &_in)
     if (workflow)
     {
       pbrMsg->set_metalness(workflow->Metalness());
-      pbrMsg->set_metalness_map(workflow->MetalnessMap());
+      pbrMsg->set_metalness_map(workflow->MetalnessMap().empty() ? "" :
```
Comment (chapulina) [inline src/Conversions.cc:262]: Are the empty checks necessary? If so, I'd suggest adding support for that case within `asFullPath`.

**Review by chapulina — APPROVED** [review]
Works for me, I tried with this model: https://app.ignitionrobotics.org/chapulina/fuel/models/Rescue%20Randy%20-%20relative%20paths

**Review by iche033 — APPROVED** [review]

**Comment by chapulina** [comment]
This is blocked by a libSDFormat8 release.

**Comment by chapulina** [comment]
@osrf-jenkins run tests, you can do it!

File: src/Conversions.cc (hunk 1 above)
Comment (nkoenig) [reply src/Conversions.cc:262]: Yeah, I added the empty checks after initially not having them.

**Comment by chapulina** [comment]
@osrf-jenkins run tests, SDFormat 8.9.0 is out!

**Pull request merged**